community/spatial-index/src/main/java/org/neo4j/gis/spatial/index/curves/SpaceFillingCurve.java

```diff
@@ -318,16 +362,17 @@ else if ( search.intersects( currentExtent ) )
     /**
      * Given a coordinate, find the corresponding normalized coordinate
      */
-    private long[] getNormalizedCoord( double[] coord )
+    long[] getNormalizedCoord( double[] coord )
     {
         long[] normalizedCoord = new long[nbrDim];
 
         for ( int dim = 0; dim < nbrDim; dim++ )
         {
             double value = clamp( coord[dim], range.getMin( dim ), range.getMax( dim ) );
-            if ( value == range.getMax( dim ) )
+            // Avoiding awkward rounding errors
+            if ( value - range.getMin( dim ) == range.getMax( dim ) - range.getMin( dim ) )
```

Comment:
It would be very hard to choose an epsilon, since we allow arbitrary small search ranges. 

This does actually help, since we are using exactly `value - range.getMin( dim )` in the else block *and* in the if statement now. The error I had before was, e.g., that `value = 1.99999999` and `max = 2` and `min = 1`, so we would choose the else block. But actually `1.99999999 - 1` would evaluate to `1`, just as `2 - 1`, so we should have chosen the if-block. If we end up with a value too big in the else block, we get off-by-one errors.